napari/utils/events/evented_model.py

```diff
@@ -276,7 +282,59 @@ def _super_setattr_(self, name: str, value: Any) -> None:
         else:
             super().__setattr__(name, value)
 
+    def _check_if_differ(self, name, old_value) -> Tuple[bool, any]:
+        """
+        Check new value of a field and emit event if it is different from the old one.
+        Returns True if event was emitted, else False.
+        """
+        new_value = getattr(self, name, object())
+        if name in self.__eq_operators__:
+            are_equal = self.__eq_operators__[name]
+        else:
+            are_equal = pick_equality_operator(new_value)
+        if not are_equal(new_value, old_value):
+            return True, new_value
+        return False, None
+
     def __setattr__(self, name: str, value: Any) -> None:
+        if name not in getattr(self, 'events', {}):
+            # This is a workaround needed because `EventedConfigFileSettings` uses
+            # `_config_path` before calling the superclass constructor
+            super().__setattr__(name, value)
+            return
+        with ComparisonDelayer(self):
+            self._primary_changes.add(name)
+            self._setattr_impl(name, value)
+
+    def _emit_pending(self):
+        if self._value_check_counter > 0 or len(self._check_queue) == 0:
```

Comment:
Generally, I find this method hard to read just from looking at the code. I don't have any specific suggestions to improve that, but a docstring or some comments could help clarify what is happening here and why.

Reply:
I have added comments. Maybe it could allow to have some suggestion. Maybe I should rename more variables.